pkg/registry/generic.go

```diff
@@ -83,6 +89,67 @@ func NewGenericClient(client *Client) *GenericClient {
 	}
 }
 
+// resolveFromIndex selects a manifest from an OCI Image Index by artifactType.
+// It returns the descriptor of the matching manifest, or an error if no match is found.
+// If no manifests have artifactType set, it falls back to checking config.mediaType
+// of each manifest to find one that matches the expected artifact type.
+func (c *GenericClient) resolveFromIndex(ctx context.Context, repo *remote.Repository, indexDesc ocispec.Descriptor, artifactType string) (ocispec.Descriptor, error) {
+	// Fetch the index manifest
+	indexData, err := content.FetchAll(ctx, repo, indexDesc)
+	if err != nil {
+		return ocispec.Descriptor{}, fmt.Errorf("unable to fetch image index: %w", err)
+	}
+
+	var index ocispec.Index
+	if err := json.Unmarshal(indexData, &index); err != nil {
+		return ocispec.Descriptor{}, fmt.Errorf("unable to parse image index: %w", err)
+	}
+
+	// First pass: look for explicit artifactType match
+	var availableTypes []string
+	var candidatesWithoutArtifactType []ocispec.Descriptor
+	for _, manifest := range index.Manifests {
+		if manifest.ArtifactType == artifactType {
+			return manifest, nil
+		}
+		if manifest.ArtifactType != "" {
+			availableTypes = append(availableTypes, manifest.ArtifactType)
+		} else {
+			// Collect manifests without artifactType for fallback check
+			candidatesWithoutArtifactType = append(candidatesWithoutArtifactType, manifest)
+		}
+	}
+
+	// Second pass: if no artifactType matches found, check config.mediaType of each manifest
+	// This handles the case where artifacts are published without explicit artifactType
+	for _, candidate := range candidatesWithoutArtifactType {
+		// Skip manifests with platform (likely container images)
+		if candidate.Platform != nil {
+			continue
+		}
+
+		// Fetch the manifest to check its config.mediaType
+		manifestData, err := content.FetchAll(ctx, repo, candidate)
+		if err != nil {
+			continue // Skip manifests we can't fetch
+		}
+
+		var manifest ocispec.Manifest
+		if err := json.Unmarshal(manifestData, &manifest); err != nil {
+			continue // Skip malformed manifests
+		}
+
+		// Check if config.mediaType matches our expected artifact type
+		if manifest.Config.MediaType == artifactType {
+			return candidate, nil
+		}
+	}
+
+	return ocispec.Descriptor{}, fmt.Errorf(
+		"no manifest with artifactType %q found in image index; available types: %v",
+		artifactType, availableTypes)
```

Comment:
Not sure if it's configurable with a linter, but I'd prefer `)` on a separate line. I think that's also more GoLang'ish style to end the last parameter with a  comma `,`